src/bridge/MatrixBanSync.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Synchronises Matrix `m.policy.rule` events with the bridge to filter specific
+ * users from using the service.
+ */
+
+import { Intent, MatrixUser, WeakStateEvent } from "matrix-appservice-bridge";
+import { MatrixGlob } from "matrix-bot-sdk";
+import { getLogger } from "../logging";
+
+const log = getLogger("MatrixBanSync");
+export interface MatrixBanSyncConfig {
+    rooms: string[];
+}
+
+enum BanEntityType {
+    Server = "m.policy.rule.server",
+    User = "m.policy.rule.user"
+}
+
+interface BanEntity {
+    matcher: MatrixGlob;
+    entityType: BanEntityType;
+    reason: string;
+}
+
+interface MPolicyContent {
+    entity: string;
+    reason: string;
+    reccomendation: "m.ban";
+}
+
+function eventTypeToBanEntityType(eventType: string): BanEntityType|null {
+    switch (eventType) {
+        case "m.policy.rule.user":
+            return BanEntityType.User;
+        case "m.policy.rule.server":
+            return BanEntityType.Server
+        default:
+            return null;
+    }
+}
+
+export class MatrixBanSync {
+    private bannedEntites = new Map<string, BanEntity>();
+    private interestingRooms = new Set<string>();
+    constructor(private config: MatrixBanSyncConfig) { }
+
+    public async syncRules(intent: Intent) {
+        this.bannedEntites.clear();
+        this.interestingRooms.clear();
+        for (const roomIdOrAlias of this.config.rooms) {
+            try {
+                const roomId = await intent.join(roomIdOrAlias);
+                this.interestingRooms.add(roomId);
+                const roomState = await intent.roomState(roomId, false) as WeakStateEvent[];
+                for (const evt of roomState) {
+                    this.handleIncomingState(evt, roomId);
+                }
+            }
+            catch (ex) {
+                log.error(`Failed to read ban list from ${roomIdOrAlias}`, ex);
+            }
+        }
+    }
+
+    /**
+     * Is the given room considered part of the bridge's ban list set.
+     * @param roomId A Matrix room ID.
+     * @returns true if state should be handled from the room, false otherwise.
+     */
+    public isInterestedInRoom(roomId: string): boolean {
```

Comment:
Ah, here's when "interesting" is a public API – but I think it makes sense here. Leaving all these comments as a food for thought though. Maybe `isTrackingRoomState()` would be more fitting here? I don't have strong opinions, just dropping it in here.